refactor(forms): mark all signal forms apis as experimental

mmalerba opened the pull request:
Ensure that each public API symbol in @angular/forms/signals is marked
as `@experimental`, also ensure that all public APIs live under `api/`.

mmalerba committed:
refactor(forms): mark all signal forms apis as experimental

Ensure that each public API symbol in @angular/forms/signals is marked
as `@experimental`

mmalerba committed:
refactor(forms): move control directive under api

Moves the control directive under api/ since it is part of the public
API. Also removes the interop abstract control from the public API

JeanMeche reviewed (approved):
reviewed-for: public-api

mmalerba committed:
refactor(forms): add validateStandardSchema to public api

Add `validateStandardSchema` which was accidentally omitted from the
public api.

mmalerba commented on packages/forms/signals/src/api/validators/standard_schema.ts:
```diff
@@ -47,6 +51,8 @@ export type IgnoreUnknownProperties<T> =
+ *
+ * @experimental 21.0.0
  */
 export function validateStandardSchema<TSchema, TValue extends IgnoreUnknownProperties<TSchema>>(
```
oh sorry, I think I forgot to add that to `validators/index.ts` at some point when moving the files around. thanks for calling it out!

mmalerba commented:
**caretaker note:** requires g3 patch update: cl/804127158

JeanMeche commented on packages/forms/signals/src/api/validators/index.ts:
```diff
@@ -13,3 +13,4 @@ export {min} from './min';
 export {minLength} from './min_length';
 export {pattern} from './pattern';
 export {required} from './required';
+export {validateStandardSchema} from './standard_schema';
```
I would expect us to also export `IgnoreUnknownProperties`/`RemoveStringIndexUnknownKey`
or replace them with opaque types in the public API.

mmalerba committed:
fixup! refactor(forms): add validateStandardSchema to public api

NothingEverHappens reviewed (approved):
reviewed-for: public-api

kirjs reviewed (approved):
reviewed-for: public-api

jelbourn reviewed (approved):
LGTM

Reviewed-for: public-api
